flake.nix

```diff
@@ -15,29 +15,33 @@
   outputs = { self, nixpkgs, mmdoc, treefmt-nix, runtimeDeps } @ args:
     let
       systems = [ "x86_64-linux" "aarch64-linux" "x86_64-darwin" "aarch64-darwin" ];
+
       eachSystem = f: nixpkgs.lib.genAttrs systems (system: f nixpkgs.legacyPackages.${system});
+
       treefmtEval = eachSystem (pkgs: treefmt-nix.lib.evalModule pkgs {
         projectRootFile = ".git/config";
-        programs.ormolu.enable = true;
+      });
+
+      buildAttrs = eachSystem (pkgs: {
+        packages = import ./pkgs/default.nix (args // { inherit (pkgs) system; });
+        devShell = pkgs.mkShell {
+          buildInputs = [
+            pkgs.git
+            pkgs.curl
+            pkgs.treefmt
+          ];
+        };
+        treefmt = treefmtEval.${pkgs.system}.config.build.check self;
       });
     in
     {
-      checks.x86_64-linux =
-        let
-          packages = nixpkgs.lib.mapAttrs' (n: nixpkgs.lib.nameValuePair "package-${n}") self.packages.x86_64-linux;
-          devShells = nixpkgs.lib.mapAttrs' (n: nixpkgs.lib.nameValuePair "devShell-${n}") self.devShells.x86_64-linux;
-        in
-        packages // devShells // {
-          treefmt = treefmtEval.x86_64-linux.config.build.check self;
-        };
-
-      formatter = eachSystem (pkgs: treefmtEval.${pkgs.system}.config.build.wrapper);
+      checks = eachSystem (pkgs: {
+        treefmt = buildAttrs.${pkgs.system}.treefmt;
+      });
```

Comment:
This isn't correct, you've removed all of the checks except for treefmt which has been changed to run on all platforms.

For CI devshells and packages need to be checks for all platforms and the treefmt check should be x86_64-linux as it was originally.